opencga-catalog/src/main/java/org/opencb/opencga/catalog/db/mongodb/FamilyMongoDBAdaptor.java

```diff
@@ -397,145 +397,161 @@ OpenCGAResult<Family> transactionalUpdate(ClientSession clientSession, Family fa
                 QueryParams.STUDY_UID.key(), QueryParams.MEMBERS.key() + "." + IndividualDBAdaptor.QueryParams.ID.key());
 
         Bson bsonQuery = parseQuery(tmpQuery);
-        return versionedMongoDBAdaptor.update(clientSession, bsonQuery, fieldsToInclude, (entrylist) -> {
-                    String familyId = entrylist.get(0).getString(QueryParams.ID.key());
-                    DataResult result = updateAnnotationSets(clientSession, studyUid, familyUid, parameters,
-                            variableSetList, queryOptions, true);
-                    List<String> familyMemberIds = family.getMembers().stream().map(Individual::getId).collect(Collectors.toList());
-                    boolean updateRoles = queryOptions.getBoolean(ParamConstants.FAMILY_UPDATE_ROLES_PARAM);
-                    boolean updatePedigree = queryOptions.getBoolean(ParamConstants.FAMILY_UPDATE_PEDIGREEE_GRAPH_PARAM);
-                    if (CollectionUtils.isNotEmpty(parameters.getAsList(QueryParams.MEMBERS.key()))) {
-                        List<Map> newIndividuals = parameters.getAsList(QueryParams.MEMBERS.key(), Map.class);
-                        Set<String> newIndividualIds = newIndividuals.stream().map(i -> (String) i.get(IndividualDBAdaptor.QueryParams.ID
-                                .key())).collect(Collectors.toSet());
-
-                        Set<String> currentIndividualIds = family.getMembers().stream().map(Individual::getId).collect(Collectors.toSet());
-
-                        // Obtain new members to be added to the family
-                        List<String> missingIndividualIds = new ArrayList<>();
-                        for (String newIndividualId : newIndividualIds) {
-                            if (!currentIndividualIds.contains(newIndividualId)) {
-                                missingIndividualIds.add(newIndividualId);
-                            }
-                        }
-
-                        // Obtain members to remove from family
-                        List<String> oldIndividualIds = new ArrayList<>();
-                        for (String currentIndividualId : currentIndividualIds) {
-                            if (!newIndividualIds.contains(currentIndividualId)) {
-                                oldIndividualIds.add(currentIndividualId);
-                            }
-                        }
 
-                        updateFamilyReferenceInIndividuals(clientSession, family, missingIndividualIds, oldIndividualIds);
-                        updateRoles = true;
-                        familyMemberIds = new ArrayList<>(newIndividualIds);
+        SnapshotVersionedMongoDBAdaptor.FunctionWithException<Family> updateFamilyReferences = (familyList) -> {
```

Comment:
This extra large lambda could have been moved to a private function, with something in the name that hints you that the function must be called from some special environment, or as a callback:

```java
private updatefamilyReferencesCallback(List<Document> familyList, ... )
```

Or even have an internal private static class to make it even clearer. This might be usefull if you wanted to add extra functionality